### PR TITLE
Better text replacement

### PR DESCRIPTION
**Pull request by jcblw**
## Description

Adds in the ability to replace a sentence with multiple of the same word. Also will throw if you attempt to replace text in a node that does not have the text.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "aracari",
-  "version": "0.1.1",
+  "version": "0.1.2",
```
Comment (jcblw) [inline package.json:3]: This is already published.

File: src/index.test.ts [hunk 2]
```diff
@@ -96,4 +96,11 @@ describe("Aracari", () => {
       `An aracari or araçari (US: /ˌɑːrəˈsɑːri/ AR-ə-SAR-ee,[1] UK: /ˌærəˈsɑːri/ ARR-ə-SAR-ee, /-ˈkɑːri/ -⁠KAR-ee)[2] is any of the medium-sized toucans that, together with the saffron toucanet, make up the genus Pteroglossus.`
     );
   });
+  test("replaceText should be to perserve text when there is two instances of the same word in a single text node", () => {
```
Comment (jcblw) [inline src/index.test.ts:99]: New functionality test

File: src/index.ts [hunk 3]
```diff
@@ -52,24 +52,27 @@ export class Aracari<T extends HTMLElement = HTMLElement> {
...
+    if (!node.textContent.match(text)) {
+      throw new Error("Text not found in node");
+    }
+    const [preText, ...postText] = node.textContent.split(text);
     const replacementNodes = [
       this.maybeCreateTextNode(preText),
       ...(Array.isArray(nodes) ? nodes : [nodes]),
-      this.maybeCreateTextNode(postText)
-    ].filter(x => x);
+      this.maybeCreateTextNode(postText.join(text)),
```
Comment (jcblw) [inline src/index.ts:74]: This is essentially the fix.

Comment (ShaunMerritt) [reply src/index.ts:74]: 🎨 
Beautiful!

**Review by ShaunMerritt — APPROVED** [review]
The ability to replace words multiple times in a node is huge!

**Comment by jcblw** [comment]
@ShaunOssa yeah I think in the future of this we probably want the option to replace at an index or replace multiple. Not sure quite how that would work though haha.

**Comment by jcblw** [comment]
@ShaunOssa also this does not replace multiple it actually allows for multiple instances of the same word in a sentence not to clobber then end of the sentence from the split. For example:

With this text node: **"112 co-sponsors of Medicare for All were on the ballot. All of them won."**

if I did a replace of **All** it would replace the text with.

**"112 co-sponsors of Medicare for *replacement* were on the ballot."**

Because the second **All** would split but the rest of the content after that would not be retained.

**Pull request merged**